[Quality Declaration] Fixed rep links and added more details to dependencies (#1116)
ros2/rclcpp: rclcpp/QUALITY_DECLARATION.md

## Dependencies [5]

Below are evaluations of each of `rclcpp`'s run-time and build-time dependencies that have been determined to influence the quality.

> It has several "buildtool" dependencies, which do not affect the resulting quality of the package, because they do not contribute to the public library API.

Reviewer: ```suggestion
It has several "buildtool" dependencies, which do not affect the resulting quality of the package, because they do not contribute to the public library API.

```

It also has several test dependencies, which do not affect the resulting quality of the package, because they are only used to build and run the test code.

> 
> 

Reviewer: ```suggestion

```

### Direct and Optional Runtime ROS Dependencies [5.i]/[5.ii]

`rclcpp` has the following runtime ROS dependencies:

#### `libstatistics_collector`

> The `libstatistics_collector` package provides a lightweight aggregation utilities to collect statistics and measure message metrics.

Reviewer: ```suggestion
The `libstatistics_collector` package provides lightweight aggregation utilities to collect statistics and measure message metrics.
```

> It is **Quality Level 4**, see its [Quality Declaration document](https://github.com/ros-tooling/libstatistics_collector/tree/master/Quality_Declaration.md).

Reviewer: ```suggestion
It is **Quality Level 4**, see its [Quality Declaration document](https://github.com/ros-tooling/libstatistics_collector/tree/master/QUALITY_DECLARATION.md).
```

#### `rcl`

> `rcl` is the ROS 2 client library in C.

Reviewer: I think `rcl` is not intented to be a client library, but a core used for the client libraries. I'd use a similar text used in its README:
```suggestion
`rcl` is a library to support implementation of language specific ROS 2 Client Libraries..
```

It is **Quality Level 4**, see its [Quality Declaration document](https://github.com/ros2/rcl/blob/master/rcl/QUALITY_DECLARATION.md).

#### `rcl_yaml_param_parser`

The `rcl_yaml_param_parser` package provides an API that is used to parse YAML configuration files which may be used to configure ROS and specific nodes.

It is **Quality Level 4**, see its [Quality Declaration document](https://github.com/ros2/rcl/tree/master/rcl_yaml_param_parser/QUALITY_DECLARATION.md).

#### `rcpputils`

> The `rcpputils` package provides an API which contains common utilities and data structures needed when programming in C++.

Reviewer: ```suggestion
The `rcpputils` package provides an API which contains common utilities and data structures useful when programming in C++.
```

> It is **Quality Level 4**, see its [Quality Declaration document](https://github.com/ros2/rcutils/blob/master/QUALITY_DECLARATION.md).

Reviewer: ```suggestion
It is **Quality Level 4**, see its [Quality Declaration document](https://github.com/ros2/rcpputils/blob/master/QUALITY_DECLARATION.md).
```

#### `rcutils`

> The `rcutils` package provides an API which contains common utilities and data structures needed when programming in C.

Reviewer: ```suggestion
The `rcutils` package provides an API which contains common utilities and data structures useful when programming in C.
```

It is **Quality Level 4**, see its [Quality Declaration document](https://github.com/ros2/rcutils/blob/master/QUALITY_DECLARATION.md).

#### `rmw`

`rmw` is the ROS 2 middleware library.

It is **Quality Level 4**, see its [Quality Declaration document](https://github.com/ros2/rmw/blob/master/rmw/QUALITY_DECLARATION.md).

#### `statistics_msgs`

The `statistics_msgs` package contains ROS 2 message definitions for reporting statistics for topics and system resources.

It is **Quality Level 4**, see its [Quality Declaration document](https://github.com/ros2/rcl_interfaces/blob/master/statistics_msgs/QUALITY_DECLARATION.md).

#### `tracetools`

> The `tracetools` package provides utilities for instrumenting the code in `rcl` so that it may be traced for debugging and performance analysis.

Reviewer: I'm not familiarized with this package, is it specific to `rcl`?
